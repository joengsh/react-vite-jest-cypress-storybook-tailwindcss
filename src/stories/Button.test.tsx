import { render } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';
import * as stories from './Button.stories';

const allStories = composeStories(stories);
describe('Button', () => {
  Object.values(allStories).forEach((Story) => {
    it('should match snapshot', () => {
      const { container } = render(<Story />);
      expect(container).toMatchSnapshot();
    });
  });
});

// should also include click test, property test including primary, size, backgroundColor, label, etc
