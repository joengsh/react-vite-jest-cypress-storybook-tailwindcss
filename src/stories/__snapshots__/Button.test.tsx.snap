// Vitest Snapshot v1

exports[`renders in the loading state 1`] = `
<div>
  <button
    class="storybook-button storybook-button--medium storybook-button--primary"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`renders in the loading state 2`] = `
<div>
  <button
    class="storybook-button storybook-button--medium storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`renders in the loading state 3`] = `
<div>
  <button
    class="storybook-button storybook-button--large storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`renders in the loading state 4`] = `
<div>
  <button
    class="storybook-button storybook-button--small storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</div>
`;
